### ISSUE TITLE
Unable to return a Promise from app.middleware

### ISSUE THREAD

**Issue by TommyBs** [actions-on-google/actions-on-google-nodejs#214]
What's the correct way to perform some kind of promise/async activity in a middleware function?

Looking at https://github.com/actions-on-google/actions-on-google-nodejs/blob/master/src/service/dialogflow/dialogflow.ts#L430 it should await for a promise if ones is received should it not?

In this contrived example

```
'use strict';

const {dialogflow} = require('actions-on-google');
const functions = require('firebase-functions');

const app = dialogflow({debug: true});
app.middleware(  (conv) => {
    return new Promise((resolve,reject) => {
        setTimeout(() => {
            conv.intent = 'resolved'
            resolve(conv);
           // also tried:
           //resolve()
          //return resolve();
         // note this will work in a normal intent handler
        }, 2000)
    })
      
});

app.fallback(conv => {
    const intent = conv.intent;
    conv.ask("Hello , your intent was changed to " + conv.intent );
//Expected 'resolved' 
//actual is whatever intent was invoked e.g 'welcome'
});

exports.dialogflowFirebaseFulfillment = functions.https.onRequest(app);

```

This actually results in an error

`TypeError: Cannot read property 'parsed' of undefined at Function.<anonymous> (/srv/node_modules/actions-on-google/dist/service/dialogflow/dialogflow.js:115:69) at Generator.next (<anonymous>) at /srv/node_modules/actions-on-google/dist/service/dialogflow/dialogflow.js:22:71 at new Promise (<anonymous>) at __awaiter (/srv/node_modules/actions-on-google/dist/service/dialogflow/dialogflow.js:18:12) at Function.handler (/srv/node_modules/actions-on-google/dist/service/dialogflow/dialogflow.js:58:16) at Object.<anonymous> (/srv/node_modules/actions-on-google/dist/assistant.js:53:32) at Generator.next (<anonymous>) at /srv/node_modules/actions-on-google/dist/assistant.js:22:71 at new Promise (<anonymous>)`

Should promises be supported by middleware? If not, can they be? I can imagine it would be quite useful to grab some data from an API etc. in middleware that is common to all requests, or to a particular user and potentially then manipulate the conv object.

**Comment by Fleker** [comment]
This was originally addressed in #151. It should be available in version 2.2.0 and up. Can you confirm that you're using a recent library version?

It may be useful to take a look at how [the tests](https://github.com/actions-on-google/actions-on-google-nodejs/blob/master/src/service/dialogflow/_test/dialogflow.test.ts#L471) are written and see if there's any code in that which may help you debug your middleware.

**Comment by TommyBs** [comment]
It looks like my package.json file had an older version in it as I'd originally imported my function from the dialogflow firebase example. Updating the version worked fine!
